src/ser/mod.rs

```diff
@@ -291,12 +316,16 @@ impl<W: io::Write> Serializer<W> {
     }
 
     fn start_indent(&mut self) -> Result<()> {
+        self.start_indent_if(|_| true)
+    }
+
+    fn start_indent_if(&mut self, condition: impl Fn(&PrettyConfig) -> bool) -> Result<()> {
```

Comment:
Why use a closure here if it only receives the config which is known ahead of time?
Maybe calling it `expand` or something would make the code more self-explanatory.

Reply:
I'm not sure what you want to rename to `expand`. The closure usage comes from the original PR ... I think the extra helper functions could be avoided since they are all just used once and enclosing them inside an if at the call site should work as well.

Reply:
You're right, that's even better!

Reply:
This should be better now. The old version also had a formatting bug, which is why I added an extra test - in compact arrays the indentation level should *not* be incremented.